### DISCUSSION TITLE
Airflow UI cannot save Slack connection properly

### DISCUSSION THREAD

**Discussion by Volatus** [apache/airflow#30402] [Q&A]
### Apache Airflow version

2.5.2

### What happened

Creating a connection in the UI with the type "Slack API" does not actually save the provided API token. (If I click "Test", the connection is working as intended, but API token will not actually be saved in the connection object, leaving the password field empty)

### What you think should happen instead

API token should persist.

### How to reproduce

This seems to only happen with the Slack API connection type, as long as you try to create one and save it, the issue should show up.

### Operating System

Docker Image based on top of the Airflow Official Image (py 3.10)

### Versions of Apache Airflow Providers

`apache-airflow-providers-slack==7.2.0`

### Deployment

Other 3rd-party Helm chart

### Deployment details

I'm using the following chart: https://github.com/airflow-helm/charts

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment]
It's not empty it's hidden from the UI for security reasons.
